projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -915,6 +915,37 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy {
         }
     }
 
+    /**
+     * Sets custom properties provided in additional template context.
+     *
+     * ```html
+     * <igx-column [templateContext]="contextObject">
+     *   <ng-template igxCell let-cell="cell">
+     *      {{ cell.column.contextObject.prop }}
+     *   </ng-template>
+     * </igx-column>
+     * ```
+     *
+     * @memberof IgxColumnComponent
+     */
+    @Input()
+    public set templateContext(value: any){
+        this._contextObject = value;
```

Comment:
1. `templateContext` suggests that the use can overwrite the context object in the column templates making it a little dubious.
Maybe `additionalTemplateContext` or something along these lines.

2. I don't think you need to separate it between a setter and a getter. Just leave it as simple property on the class.
3. It will be nice to access the additional context directly from the template binding:
```html
<ng-template igxCell let-props="additionalContext">
...
</ng-template>
```